clarity/contracts/alex-vault.clar

```diff
@@ -32,26 +42,86 @@
   (contract-call? token get-balance (as-contract tx-sender))
 )
 
+;; if sender is an approved contract, then transfer requested amount :qfrom vault to recipient
+(define-public (transfer-ft (token <ft-trait>) (amount uint) (sender principal) (recipient principal))
+  (begin     
+    (asserts! (default-to false (get can-transfer (map-get? approved-contracts { name: sender }))) ERR-NOT-AUTHORIZED)
```

Comment:
Can be simplified if the map is simplified.

```clojure
(asserts! (default-to false (map-get? approved-contracts sender)) ERR-NOT-AUTHORIZED)
```

Reply:
Also, if there is a lot of re-use, one can consider turning it into a helper function.

```clojure
(define-private (check-is-approved (sender principal))
  (ok (asserts! (default-to false (map-get? approved-contracts sender)) ERR-NOT-AUTHORIZED))
)
```

Usage:

```clojure
(try! (check-is-approved sender))
```